frontend/app/components/LeafletMap.js

```diff
@@ -212,10 +212,10 @@ export default class LeafletContainer extends Component {
 
             case Types.Polygon:
                 /* TODO: we only support normal polygons w/o holes and multipolygons okay? */
-                let points = data;/*new Array(data.length);
+                let points = new Array(data.length);
                 for(let i = 0; i < data.length; i++) {
-                    points.push([ data[i][0], data[i][1] + shift ]);
-                }*/
+                    points[i] = [ data[i][0], data[i][1] + shift ];
+                }
                 shape = Leaflet.polygon(points, options);
```

Comment:
Push was a non-working code which I couldn't test because Polygon wasn't available at the time. I would pre-create an array of points _and_ push new ones resulting in 2N amount of points of which first N are undefined. You commented it out at some point.